contracts/Company/Extras/ExitController.sol

```diff
@@ -0,0 +1,277 @@
+            equityTokens = EQUITY_TOKEN.balanceOfAt(investor, _manualPayoutResolutionStart);
+        }
+        proceeds = eligibleProceedsForTokens(equityTokens);
+        return (equityTokens, proceeds);
+    }
+
+    //
+    // IERC223TokenCallback (exit proceeds disbursal)
+    //
+
+    /// allows contract to receive and distribute proceeds
+    /// this can only be done in the funded state
+    function tokenFallback(address from, uint256 amount, bytes)
+        public
+    {
+        require(amount > 0, "NF_ZERO_TOKENS");
+
+        // if we're in the setup state, this contract is waiting
+        // for the nominee to send the exit funds
+        if (_state == State.Setup) {
+            // we only allow eurotokens for this operation
+            require(msg.sender == address(EURO_TOKEN), "NF_ETO_INCORRECT_TOKEN");
+            // only the nominee may send proceeds to this contract
+            require(from == EQUITY_TOKEN.nominee(), "NF_ONLY_NOMINEE");
+            // start the payout
+            startPayout();
+        }
+        // when we already are in the closing state, investors can send
+        // their tokens to  be burned and converted to euro token
+        else if ( _state == State.Payout ) {
+            // now we only allow conversion of the tokens into neumarks
+            require(msg.sender == address(EQUITY_TOKEN), "NF_ETO_UNK_TOKEN");
+            // investor must have sent all of his tokens
+            require(EQUITY_TOKEN.balanceOf(from) == 0, "NF_MUST_SEND_ALL_TOKENS");
+            // payout exit proceeds
+            payExitProceeds(from, amount);
+        } else {
+            revert("UNEXPECTED_OPERATION");
+        }
+    }
+
+    function startManualPayoutResolution()
+        public
+    {
+        require(_state == State.Payout, "NF_INCORRECT_STATE");
+        require(msg.sender == EQUITY_TOKEN.nominee(), "NF_ONLY_NOMINEE");
+        transitionTo(State.ManualPayoutResolution);
+        _manualPayoutResolutionStart = EQUITY_TOKEN.currentSnapshotId();
+    }
+
+    function payoutManually(address lostWallet, address newWallet)
+        public
+    {
+        // only in manual payout state
+        require(_state == State.ManualPayoutResolution, "NF_INCORRECT_STATE");
+        // only the nominee may do manual payouts
+        require(msg.sender == EQUITY_TOKEN.nominee(), "NF_ONLY_NOMINEE");
+        // we need a valid receiver address
+        require(newWallet != 0x0, "NF_INVALID_NEW_WALLET");
+        // we can only process wallets that have not been manually resolved yet
+        require(payoutManuallyResolved[lostWallet] == false, "NF_ALREADY_PAYED_OUT");
+        // only allow when timestamp has moved to next date
+        require(_manualPayoutResolutionStart < EQUITY_TOKEN.currentSnapshotId(), "NF_WAIT");
+
+        (uint256 _tokens, uint256 _proceeds) = eligibleProceedsForInvestor(lostWallet);
+        require(_proceeds > 0, "NF_NO_PROCEEDS");
+
```

Comment:
Maybe it makes sense to check claims of `newWallet` and add requirement for KYC and bank account? It will help mitigate manual mistakes from nominee that will send those transactions.

Reply:
I would say this is already done in the euro token controller contract in     function isTransferAllowedPrivate(address from, address to, bool allowPeerTransfers). So wallets that are not verified can't receive our-t. Could you double check if you think this is the case too?